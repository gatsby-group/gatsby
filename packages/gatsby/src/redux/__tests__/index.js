const path = require(`path`)
const { saveState, store } = require(`../index`)
const { writeToCache } = require(`../persist`)
const {
  actions: { createPage },
} = require(`../actions`)

jest.mock(`../persist`)

describe(`redux db`, () => {
  beforeEach(() => {
    store.dispatch(
      createPage(
        {
          path: `/my-sweet-new-page/`,
          component: path.resolve(`./src/templates/my-sweet-new-page.js`),
          // The context is passed as props to the component as well
          // as into the component's GraphQL query.
          context: {
            id: `123456`,
          },
        },
        { name: `default-site-plugin` }
      )
    )

    writeToCache.mockClear()
  })
  it(`should write cache to disk`, async () => {
    await saveState()

    expect(writeToCache).toBeCalled()
  })

  it(`does not write to the cache when DANGEROUSLY_DISABLE_OOM is set`, async () => {
    process.env.DANGEROUSLY_DISABLE_OOM = true

    await saveState()

    expect(writeToCache).not.toBeCalled()

    delete process.env.DANGEROUSLY_DISABLE_OOM
  })
})
