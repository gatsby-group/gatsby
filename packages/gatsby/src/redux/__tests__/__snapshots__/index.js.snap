// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux db should write cache to disk 1`] = `
Object {
  "componentDataDependencies": Object {
    "connections": Object {},
    "nodes": Object {},
  },
  "components": Map {
    "/Users/username/dev/site/src/templates/my-sweet-new-page.js" => Object {
      "componentPath": "/Users/username/dev/site/src/templates/my-sweet-new-page.js",
      "isInBootstrap": true,
      "pages": Set {
        "/my-sweet-new-page/",
      },
      "query": "",
    },
  },
  "pages": Map {
    "/my-sweet-new-page/" => Object {
      "component": "/Users/username/dev/site/src/templates/my-sweet-new-page.js",
      "componentChunkName": "component---users-username-dev-site-src-templates-my-sweet-new-page-js",
      "componentPath": "/Users/username/dev/site/src/templates/my-sweet-new-page.js",
      "context": Object {
        "id": "123456",
      },
      "internalComponentName": "ComponentMySweetNewPage",
      "isCreatedByStatefulCreatePages": undefined,
      "matchPath": undefined,
      "path": "/my-sweet-new-page/",
      "pluginCreatorId": undefined,
      "pluginCreator___NODE": undefined,
      "updatedAt": 1571231761339,
    },
  },
  "staticQueryComponents": Map {},
  "status": Object {
    "plugins": Object {},
  },
  "webpackCompilationHash": "",
}
`;
